sdk/aqueduct/enums.py

```diff
@@ -128,8 +128,24 @@ class DisplayNodeType(str, Enum, metaclass=MetaEnum):
     ARTIFACT = "ARTIFACT"
 
 
-class ArtifactType(Enum, metaclass=MetaEnum):
-    TABLE = "table"
-    NUMBER = "number"
+class ArtifactType(str, Enum, metaclass=MetaEnum):
+    UNTYPED = "untyped"
+    STRING = "string"
     BOOL = "boolean"
-    PARAM = "param"
+    NUMERIC = "numeric"
+    DICT = "dictionary"
+    TUPLE = "tuple"
+    TABULAR = "tabular"
+    JSON = "json"
+    BYTES = "bytes"
+    IMAGE = "image"
+    PICKLABLE = "picklable"
+
+class SerializationType(str, Enum, metaclass=MetaEnum):
+    TABULAR = "tabular"
+    JSON = "json"
+    PICKLE = "pickle"
+    IMAGE = "image"
+    STANDARD = "standard"
+    BYTES = "bytes"
+    
```

Comment:
It's not clear to me as a reader what the relationship between artifact type and serialization type is. Is it a many-to-many mapping? Is there only one possible serialization for each type? Can we make this explicit to a reader with a map of [ArtifactType] -> List[SerializationType] denoting what serializations we support for each type? In addition to being useful documentation, we can use this map as validation before sending stuff to the backend too.

Reply:
Each artifact type corresponds to exactly one serialization method. We aren't sending these to the backend right? It's the other way around: the backend is sending the preview response to the SDK, with the type info as a hint so that SDK knows which serialization method to use. The map you suggested is still useful because we can replace the `if-else` in `_deserialize_content` function of the untyped artifact class with this map.

Reply:
right, and I think it would help a lot as a "documentation in code" especially. Can we also have something similar on the go side?